CPE200Lab1/CPE200Lab1/MainForm.cs

```diff
@@ -200,32 +70,9 @@ private void btnClear_Click(object sender, EventArgs e)
 
         private void btnBack_Click(object sender, EventArgs e)
         {
-            if (lblDisplay.Text is "Error")
-            {
-                return;
-            }
-            if (isAfterEqual)
-            {
-                return;
-            }
-            if (!isAllowBack)
-            {
-                return;
-            }
-            if(lblDisplay.Text != "0")
-            {
-                string current = lblDisplay.Text;
-                char rightMost = current[current.Length - 1];
-                if(rightMost is '.')
-                {
-                    hasDot = false;
-                }
-                lblDisplay.Text = current.Substring(0, current.Length - 1);
-                if(lblDisplay.Text is "" || lblDisplay.Text is "-")
-                {
-                    lblDisplay.Text = "0";
-                }
-            }
-        }
+			engine.removeLastDigit();
```

Comment:
Removing last digit should be interface code, not logic (i.e., engine) code. As you can see, it doesn't change any state inside engine. It will be nonsense to put anything that will not change engine's state in engine code.